animations/completed_scenarios/lib/02_repeat_proposed.dart

```diff
@@ -28,54 +28,36 @@ class RepeatScenario extends StatefulWidget {
 
 class RepeatScenarioState extends State<RepeatScenario>
     with SingleTickerProviderStateMixin {
-  late AnimationController _controller;
-  late Animation<double> _opacityAnimation;
-
   bool _isRepeating = false;
 
-  @override
-  void initState() {
-    super.initState();
-    _controller = AnimationController(
-      duration: const Duration(seconds: 2),
-      vsync: this,
-    );
-    _opacityAnimation = Tween<double>(
-      begin: 0.0,
-      end: 1.0,
-    ).animate(_controller);
-  }
-
-  @override
-  void dispose() {
-    _controller.dispose();
-    super.dispose();
-  }
-
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(title: Text('Repeat Scenario')),
       body: Center(
         child: InheritedAnimationController(
-          child: FadeTransition(
-            opacity: InheritedAnimationController.of(context).controller,
-            child: ElevatedButton(
-              onPressed: () {
-                setState(() {
-                  if (_isRepeating) {
-                    InheritedAnimationController.of(context).controller.stop();
-                    _isRepeating = false;
-                  } else {
-                    _isRepeating = true;
-                    InheritedAnimationController.of(
-                      context,
-                    ).controller.repeat(reverse: true);
-                  }
-                });
-              },
-              child: Text('Fade in and out'),
-            ),
+          child: Builder(
```

Comment:
This is unfortunate, and it might make this scenario difficult to complete. We might want to factor out the `FadeTransition` and it's subtree into a separate widget, so that the correct `BuildContext` is available (the BuildContext needs to find the `InheritedAnimationController` when you call `of()`)

Reply:
I went ahead and factored this out in [ff2c6fe](https://github.com/flutter/uxr/pull/153/commits/ff2c6feddfb5f1d4ac99c24e30aa40b1983ef692)